rag_experiment_accelerator/evaluation/eval.py

```diff
@@ -546,68 +541,75 @@ def compute_metrics(
-        score = fuzzy(actual, expected)
-    elif metric_type == "bert_all_MiniLM_L6_v2":
-        all_MiniLM_L6_v2 = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
-        score = compare_semantic_document_values(actual, expected, all_MiniLM_L6_v2)
-    elif metric_type == "bert_base_nli_mean_tokens":
-        base_nli_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-base-nli-mean-tokens"
-        )
-        score = compare_semantic_document_values(actual, expected, base_nli_mean_tokens)
-    elif metric_type == "bert_large_nli_mean_tokens":
-        large_nli_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-large-nli-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, large_nli_mean_tokens
-        )
-    elif metric_type == "bert_large_nli_stsb_mean_tokens":
-        large_nli_stsb_mean_tokens = SentenceTransformer(
-            "sentence-transformers/bert-large-nli-stsb-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, large_nli_stsb_mean_tokens
-        )
-    elif metric_type == "bert_distilbert_base_nli_stsb_mean_tokens":
-        distilbert_base_nli_stsb_mean_tokens = SentenceTransformer(
-            "sentence-transformers/distilbert-base-nli-stsb-mean-tokens"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, distilbert_base_nli_stsb_mean_tokens
-        )
-    elif metric_type == "bert_paraphrase_multilingual_MiniLM_L12_v2":
-        paraphrase_multilingual_MiniLM_L12_v2 = SentenceTransformer(
-            "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"
-        )
-        score = compare_semantic_document_values(
-            actual, expected, paraphrase_multilingual_MiniLM_L12_v2
-        )
-    elif metric_type == "llm_answer_relevance":
-        score = llm_answer_relevance(response_generator, question, actual)
-    elif metric_type == "llm_context_precision":
-        score = llm_context_precision(response_generator, question, retrieved_contexts)
-    elif metric_type == "llm_context_recall":
-        score = llm_context_recall(
-            response_generator, question, expected, retrieved_contexts
-        )
-    else:
-        pass
+    match metric_type:
+        case "lcsstr":
+            score = lcsstr(actual, expected)
+        case "lcsseq":
+            score = lcsseq(actual, expected)
+        case "cosine":
+            score = cosine(actual, expected)
+        case "jaro_winkler":
+            score = jaro_winkler(actual, expected)
+        case "hamming":
+            score = hamming(actual, expected)
+        case "jaccard":
+            score = jaccard(actual, expected)
+        case "levenshtein":
+            score = levenshtein(actual, expected)
+        case "fuzzy":
+            score = fuzzy(actual, expected)
+        case "bert_all_MiniLM_L6_v2":
```

Comment:
First of all, why these models are hardcoded?
Second of all, why not create them inside metric function for unification?
Or move them to the config?

This is very mysterious code, but even if there is a meaning into hardcoded names - it still can be generalized in the same way as previous!

Just move these names into evaluations/transformer_based_metrics.py (or some other name, I suck at naming) and do the same trick as the first one. So you will be able to check if you can import name from evaluations/metrics.py and if you can't try import it from evaluations/transformer_based_metrics.py and if you can't try import it from the next one and so on.

Reply:
Split eval.py into modules now.